File: employeeImportance.py
"""
# Definition for Employee.
class Employee:
    def __init__(self, id: int, importance: int, subordinates: List[int]):
        self.id = id
        self.importance = importance
        self.subordinates = subordinates
"""
"""
// Time Complexity : O(N)
// Space Complexity : O(N)
// Leetcode passes: yes
// Any problems while solving this problem : no
"""

class Solution:
    def getImportance(self, employees: List['Employee'], id: int) -> int:
        
        output = 0
        graph = {}
        for employee in employees:
            graph[employee.id] = employee
            
        queue = []
        queue.append(id)
        
        while queue:
            
            curr = queue.pop(0)
            output += graph[curr].importance
            for j in graph[curr].subordinates:
                queue.append(j)
                
        return output
        

File: employeeImportance.js
// Time Complexity : O(V+E)
// Space Complexity : O(V+E)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No

var GetImportance = function(employees, id) {
    if(employees == null || employees.length == 0) return 0;
    let map = new Map();
    employees.forEach(({id, importance, subordinates}) => {
        map[id] = {importance, subordinates}
    })
    
    let q = [];
    q.push(id)
    let res = 0;
    
    while(q.length > 0){
        let currEmp = map[q.shift()]
        res += currEmp.importance
            q.push(...currEmp.subordinates)
    } 
    return res   
};